commands/delete-database.go

```diff
@@ -15,81 +17,177 @@ func init() {
+	}
 
-	// get all database names
-	names, err := res.DB.Session.DatabaseNames()
+	if bulk {
+		dbs := res.MetaDB.GetDatabases()
+		copy(names, dbs)
+	} else if match != "" {
 
-	// check if database exists
-	dbExists := util.StringInSlice(database, names)
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
 
-	// check if metadatabase record for database exists
-	mDBExists := util.StringInSlice(database, res.MetaDB.GetDatabases())
+		// Find dbs with matching names
+		for _, db := range dbs {
+			if strings.Contains(db, match) {
+				names = append(names, db)
+			}
+		}
 
-	// if no force flag, verify action with the user
-	if !forceFlag {
-		fmt.Print("Are you sure you want to delete database ", database, " [y/N] ")
+	} else if regex != "" {
 
-		read := bufio.NewReader(os.Stdin)
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
 
-		response, _ := read.ReadString('\n')
+		// Compile regex, check if it's valid
+		regq, err := regexp.Compile(regex)
+		if err != nil {
+			return cli.NewExitError(err.Error, -1)
+		}
 
-		response = strings.ToLower(strings.TrimSpace(response))
-		if response == "y" || response == "yes" {
-			fmt.Println("Deleting database: ", database)
-		} else {
-			return cli.NewExitError("Database "+database+" was not deleted.", 0)
+		// Find dbs with regex query
+		for _, db := range dbs {
+			matched := regq.MatchString(db)
+			if matched {
+				names = append(names, db)
+			}
+		}
+
+	} else {
+		// Single database deletion
+		tgtDB := c.Args().Get(0)
+		// get all database names
+		dbs := res.MetaDB.GetDatabases()
+		if find(dbs, tgtDB) != len(dbs) {
+			names = append(names, tgtDB)
 		}
+
 	}
 
-	// delete database if it exists
-	if dbExists {
-		if res.DB.Session.DB(database).DropDatabase() != nil {
-			return cli.NewExitError("Failed to delete database", -1)
+	// check if we have databases
+	if len(names) == 0 {
+		return cli.NewExitError("Failed to find any databases", -1)
+	}
+
+	// if no force or dry run flag, verify action
+	if !force && !dryRun {
+		if confirmAction("Confirm we'll be deleting the following databases:\n" + strings.Join(names, "\n")) {
+			fmt.Println("Deleting databases...")
+		} else {
+			return cli.NewExitError("Nothing deleted, no changes have been made", 0)
 		}
 	}
 
-	// remove metadatabase record if it exists
-	if mDBExists {
-		if res.MetaDB.DeleteDB(database) != nil {
-			return cli.NewExitError("Failed to update metadb", -1)
+	// Iterate through databases to delete and delete them one by one
+	for _, database := range names {
+		dberr := deleteSingleDatabase(res, names, database, dryRun)
+		if dberr != nil {
+			return cli.NewExitError(dberr.Error, -1)
 		}
 	}
 
-	// check if we have databases
-	if err != nil || len(names) == 0 {
-		return cli.NewExitError("Failed to find any databases", -1)
+	// Dry run warning
+	if dryRun {
+		fmt.Fprintf(os.Stdout, "\t[-] This was a dry run of the delete command, nothing has been changed!\n")
```

Comment:
Can you switch all these `fmt.Fprintf(os.Stdout` to just be `fmt.Printf(`?  I think they were already in the file that way, but I don't know of a good reason to keep them and it's inconsistent with the Printf and Println we've been using lately.